src/native/libs/System.IO.Compression.Native/entrypoints.c

```diff
@@ -5,24 +5,13 @@
 
 // Include System.IO.Compression.Native headers
 #include "pal_zlib.h"
-#include <external/brotli/include/brotli/decode.h>
-#include <external/brotli/include/brotli/encode.h>
-#include <external/brotli/include/brotli/port.h>
-#include <external/brotli/include/brotli/types.h>
+#include <brotli/decode.h>
+#include <brotli/encode.h>
+#include <brotli/port.h>
+#include <brotli/types.h>
 
 static const Entry s_compressionNative[] =
 {
-    DllImportEntry(BrotliDecoderCreateInstance)
-    DllImportEntry(BrotliDecoderDecompress)
-    DllImportEntry(BrotliDecoderDecompressStream)
-    DllImportEntry(BrotliDecoderDestroyInstance)
-    DllImportEntry(BrotliDecoderIsFinished)
-    DllImportEntry(BrotliEncoderCompress)
-    DllImportEntry(BrotliEncoderCompressStream)
-    DllImportEntry(BrotliEncoderCreateInstance)
-    DllImportEntry(BrotliEncoderDestroyInstance)
-    DllImportEntry(BrotliEncoderHasMoreOutput)
-    DllImportEntry(BrotliEncoderSetParameter)
```

Comment:
`BrotliDecoderCreateInstance` is used in initialization of brotli stream. Do these tests pass in your environment without these: https://github.com/dotnet/runtime/blob/1d08e1/src/libraries/System.IO.Compression.Brotli/tests/CompressionStreamUnitTests.Brotli.cs?

Reply:
Yes, they do:

```
$ head ./artifacts/bin/System.IO.Compression.Brotli.Tests/Debug/net7.0-unix/testResults.xml
<?xml version="1.0" encoding="utf-8"?>
<assemblies timestamp="03/10/2022 20:28:10">
  <assembly name="System.IO.Compression.Brotli.Tests.dll" environment="64-bit .NET  [collection-per-class, parallel (8 threads)]" test-framework="xUnit.net 2.4.2.0" run-date="2022-03-10" run-time="20:27:28" total="244" passed="244" failed="0" skipped="0" time="41.938" errors="0">
    <errors />
    <collection total="214" passed="214" failed="0" skipped="0" name="Test collection for System.IO.Compression.BrotliStreamUnitTests" time="29.320">
      <test name="System.IO.Compression.BrotliStreamUnitTests.Ctor_DisposedBaseStream(mode: Compress)" type="System.IO.Compression.BrotliStreamUnitTests" method="Ctor_DisposedBaseStream" time="0.2309207" result="Pass" />
      <test name="System.IO.Compression.BrotliStreamUnitTests.Ctor_DisposedBaseStream(mode: Decompress)" type="System.IO.Compression.BrotliStreamUnitTests" method="Ctor_DisposedBaseStream" time="0.0029734" result="Pass" />
      <test name="System.IO.Compression.BrotliStreamUnitTests.WrapStreamReturningBadReadValues" type="System.IO.Compression.BrotliStreamUnitTests" method="WrapStreamReturningBadReadValues" time="0.275656" result="Pass" />
      <test name="System.IO.Compression.BrotliStreamUnitTests.Disposed_ThrowsObjectDisposedException" type="System.IO.Compression.BrotliStreamUnitTests" method="Disposed_ThrowsObjectDisposedException" time="0.3359914" result="Pass" />
      <test name="System.IO.Compression.BrotliStreamUnitTests.ZeroByteRead_PerformsZeroByteReadOnUnderlyingStreamWhenDataNeeded(mode: SyncArray)" type="System.IO.Compression.BrotliStreamUnitTests" method="ZeroByteRead_PerformsZeroByteReadOnUnderlyingStreamWhenDataNeeded" time="0.0407726" result="Pass" />
...
```

Can you help me understand why you thought they wouldn't pass? .NET loads up `libSystem.IO.Compression.Native.so`, and that's linked to the system libbrotlienc.so/libbrotlidec.so, so all the `BrotliDecoderCreateInstance` and similar symbols are still available. Unless ... there are private symbols that entrypoints.c was exporting before?

Reply:
On Ubuntu, I see libbrotli has `BrotliDecoderCreateInstance` property exported as text symbol:
```sh
$ apt install -y libbrotli-dev
$ nm -D /usr/lib/aarch64-linux-gnu/libbrotlidec.so.1.0.7 
0000000000005028 T BrotliDecoderCreateInstance
0000000000006048 T BrotliDecoderDecompress
0000000000005148 T BrotliDecoderDecompressStream
0000000000005100 T BrotliDecoderDestroyInstance
0000000000006280 T BrotliDecoderErrorString
0000000000006278 T BrotliDecoderGetErrorCode
0000000000006120 T BrotliDecoderHasMoreOutput
0000000000006248 T BrotliDecoderIsFinished
0000000000006228 T BrotliDecoderIsUsed
0000000000004fc8 T BrotliDecoderSetParameter
0000000000006160 T BrotliDecoderTakeOutput
0000000000006410 T BrotliDecoderVersion
                 U BrotliDefaultAllocFunc
                 U BrotliDefaultFreeFunc
                 U BrotliGetDictionary
                 U BrotliGetTransforms
                 U BrotliTransformDictionaryWord
                 w _ITM_deregisterTMCloneTable
                 w _ITM_registerTMCloneTable
                 w __cxa_finalize
                 w __gmon_start__
                 U __stack_chk_fail
                 U __stack_chk_guard
                 U _kBrotliContextLookupTable
                 U _kBrotliPrefixCodeRanges
                 U free
                 U malloc
                 U memcpy
                 U memset
```

Perhaps this condition is failing on fedora build agent: https://github.com/dotnet/runtime/blob/25facd719ef8c86f358bdd80e0b779bce25b1bd3/src/native/external/brotli/include/brotli/port.h#L237
which is why it is picking up empty `BROTLI_PUBLIC` definition from line 247 (and resulting in `U` which you are seeing, instead of `T` in `nm` output)?

Reply:
Fedora has the same output for `nm -D libbrotlidec`:

```
$ nm -D /usr/lib64/libbrotlidec.so.1.0.9 
00000000000064a0 T BrotliDecoderCreateInstance
0000000000007b40 T BrotliDecoderDecompress
0000000000006750 T BrotliDecoderDecompressStream
00000000000066e0 T BrotliDecoderDestroyInstance
0000000000007f40 T BrotliDecoderErrorString
0000000000007f30 T BrotliDecoderGetErrorCode
0000000000007dd0 T BrotliDecoderHasMoreOutput
0000000000007f00 T BrotliDecoderIsFinished
0000000000007ee0 T BrotliDecoderIsUsed
0000000000006430 T BrotliDecoderSetParameter
0000000000007e10 T BrotliDecoderTakeOutput
0000000000008140 T BrotliDecoderVersion
                 U BrotliDefaultAllocFunc
                 U BrotliDefaultFreeFunc
                 U BrotliGetDictionary
                 U BrotliGetTransforms
                 U BrotliTransformDictionaryWord
                 w __cxa_finalize@GLIBC_2.2.5
                 w __gmon_start__
                 w _ITM_deregisterTMCloneTable
                 w _ITM_registerTMCloneTable
                 U _kBrotliContextLookupTable
                 U _kBrotliPrefixCodeRanges
                 U malloc@GLIBC_2.2.5
                 U memcpy@GLIBC_2.14
                 U memmove@GLIBC_2.2.5
                 U memset@GLIBC_2.2.5
                 U __stack_chk_fail@GLIBC_2.4
```

The `U` was for symbols `entrypoints.c` was using, but are not defined in `entrypoints.c` - they are provided by the brotli libraries